src/librustc/lint/builtin.rs

```diff
@@ -224,6 +224,12 @@ declare_lint! {
 }
 
 declare_lint! {
+    pub LEGACY_CONSTRUCTOR_VISIBILITY,
+    Deny,
```

Comment:
The policy is usually warn-by-default, then deny-by-default, no? But we can make an exception, though we should make an RFC bot request about it. Either way, we should contact the authors of the crates in question and/or submit PRs.